### PR TITLE
Make assertion selector control fading of non selected spans

### PR DESCRIPTION
**Pull request by jorgeepc**
This PR implements the following features:
- When `creating/editing` an assertion, we fade out the non selected spans so the user can see the impact of their selector change.
- When `clicking` on a particular assertion block in a test result, we go into 'selected' mode. This outlines the assertion block and fade out the unselected spans in the graph and the timeline view.

## Changes

- Add new `affectedSpans` redux state.
- Update `Graph View` and `Timeline View` to handle the affected spans.

## Fixes

- #567 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

## Loom

https://www.loom.com/share/9b4764b3705e4100bfac080a9821bd45

## Coverage

Screen Shot 2022-05-30 at 16 17 04

### CODE REVIEW

File: web/src/components/AssertionCard/AssertionCard.styled.ts [hunk 1]
```diff
@@ -2,12 +2,13 @@ import {DeleteOutlined, EditOutlined} from '@ant-design/icons';
 import {Tag, Typography} from 'antd';
 import styled, {css} from 'styled-components';
 
-export const AssertionCard = styled.div`
+export const AssertionCard = styled.div<{isSelected: boolean}>`
```
Comment (cescoferraro) [inline web/src/components/AssertionCard/AssertionCard.styled.ts:5]: use transient props
https://styled-components.com/docs/api#transient-props

File: web/src/components/Diagram/components/TimelineChart.styled.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
-import styled from 'styled-components';
+import styled, {css} from 'styled-components';
 
-export const Container = styled.div<{barHeight: number}>`
+export const Container = styled.div<{barHeight: number; showAffected: boolean}>`
```
Comment (cescoferraro) [inline web/src/components/Diagram/components/TimelineChart.styled.ts:3]: use transient props
https://styled-components.com/docs/api#transient-props

**Review by cescoferraro — CHANGES_REQUESTED** [review]
Looks great! Added a few comments to avoid console warnings

File: web/src/components/AssertionCard/AssertionCard.styled.ts (hunk 1 above)
Comment (jorgeepc) [reply web/src/components/AssertionCard/AssertionCard.styled.ts]: Done ✅

File: web/src/components/Diagram/components/TimelineChart.styled.ts (hunk 2 above)
Comment (jorgeepc) [reply web/src/components/Diagram/components/TimelineChart.styled.ts]: Done ✅

File: web/src/models/AssertionResults.model.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ const AssertionResults = ({allPassed = false, results = []}: TRawAssertionResult
     resultList: results.map(({selector = '', results: resultList = []}) => ({
       id: uniqueId(),
       selector,
-      spanCount: AssertionService.getSpanCount(resultList),
+      spanIds: AssertionService.getSpanIds(resultList),
```
Comment (xoscar) [inline web/src/models/AssertionResults.model.ts:13]: 👍

File: web/src/redux/slices/TestDefinition.slice.ts [hunk 4]
```diff
@@ -74,6 +76,12 @@ const testDefinitionSlice = createSlice({
     setAssertionResults(state, {payload}: PayloadAction<TAssertionResults>) {
       state.assertionResults = payload;
     },
+    clearAffectedSpans(state) {
```
Comment (xoscar) [inline web/src/redux/slices/TestDefinition.slice.ts:79]: Maybe a change for the future but, I was thinking of decoupling both the affected spans and the selected span into a separated slice. As you can see, the setSelectedSpan function is being drilled into multiple component levels. We can solve it by having a central place for both.

**Review by xoscar — APPROVED** [review]
GJ @jorgeepc! I left some non-blocking suggestions. Let me know what you think

File: web/src/redux/slices/TestDefinition.slice.ts (hunk 4 above)
Comment (jorgeepc) [reply web/src/redux/slices/TestDefinition.slice.ts:82]: Yeah, like the idea. I can work on that after the MVP priorities.

**Pull request merged**